graph-gateway/src/main.rs

```diff
@@ -269,15 +271,18 @@ async fn main() {
     spawn(async move {
         let router = Router::new().route("/metrics", routing::get(handle_metrics));
 
-        Server::bind(&SocketAddr::new(
+        let metrics_listener = TcpListener::bind(SocketAddr::new(
             IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
             metrics_port,
         ))
-        // disable Nagel's algorithm
-        .tcp_nodelay(true)
-        .serve(router.into_make_service())
         .await
-        .expect("Failed to start metrics server");
+        .expect("Failed to bind metrics server");
+        axum::serve(metrics_listener, router.into_make_service())
+            // Wait until https://github.com/tokio-rs/axum/pull/2653 is released
```

Comment:
Sometimes I grep for "TODO".
```suggestion
            // TODO: Wait until https://github.com/tokio-rs/axum/pull/2653 is released
```